src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx

```diff
@@ -93,6 +85,12 @@ const DownloadModal = ({
     setFastaSelected(!isFastaSelected);
   };
 
+  const handleCloseModal = () => {
+    setFastaSelected(false);
```

Comment:
I might not be understanding how the DownloadModal works, but does the DownloadModal unmount with the `onClose` call? If so, I'm unsure about the benefit of calling these set_Selected(false) here. Seems like the unmount will inherently set them false by just knocking the whole component into oblivion?

That said, I don't actually know the larger use, if you think it makes sense, disregard what I'm saying. Also, heck, I can even argue against myself as this being a form of future-proofing if the onClose unmount stops happening but we would still want the selections to false out.

Reply:
I think the component does not actually unmount. The parent just tells the component whether or not to display itself. So, we show the component only when parent says `open=true`, but the component itself does not get destroyed, so it maintains the state.

Would be a different story if it were like
```
// ParentComponent/index.tsx

...
if (someParentState) return null;
return <DownloadModal />
```